### PR TITLE
#363 fix - trigger onWarning with invalid yaml

### PR DESCRIPTION
**Pull request by rezaa91**
fix for: https://github.com/paazmaya/yaml-validator/issues/363

Also, updated regex for detecting line number in js-yaml error

### CODE REVIEW

**Comment by rezaa91** [comment]
Are we able to ignore the sonar report? I don't think there is a risk with this regex

File: package.json [hunk 1]
```diff
@@ -44,6 +44,7 @@
     "eslint-config-paazmaya": "8.0.0",
     "eslint-plugin-node": "11.1.0",
     "nyc": "15.1.0",
+    "sinon": "^14.0.2",
```
Comment (paazmaya) [inline package.json:47]: Please pin the version

Comment (rezaa91) [reply package.json]: version pinned to 14.0.2

**Comment by paazmaya** [comment]
Looks good and will get merged, but the conflict in lock file needs to be solved.
Moved from `nyc` to `c8`, which causes it...

**Comment by rezaa91** [comment]
resolved lock conflicts @paazmaya

**Comment by paazmaya** [comment]
Thank you for your contribution @rezaa91 !

**Pull request merged**